app/src/main/java/com/mensinator/app/settings/SettingsViewModel.kt

```diff
@@ -203,9 +208,14 @@ class SettingsViewModel(
         _viewState.update { it.copy(showExportDialog = show) }
     }
 
-    fun handleImport(importPath: String) {
+    fun handleImport(importPath: String, source: ImportSource) {
         try {
-            exportImport.importDatabase(importPath)
+            when (source) {
+                ImportSource.MENSINATOR -> exportImport.importDatabase(importPath)
+                ImportSource.CLUE -> clueImport.importFileToDatabase(importPath)
+                ImportSource.FLO -> floImport.importFileToDatabase(importPath)
+            }
+
             Toast.makeText(
```

Comment:
This success toast is always run, even when the import ran into an error.

It's possible to either have the `importFileToDatabase` methods
- throw an exception to signal that something was wrong
- return a `Boolean` on whether it was successful or not

Imo in this case we should go for a Boolean. It's not recoverable, we can't differentiate between errors anyhow.

Reply:
To test this, I selected Clue and imported a Mensinator json.